### PR TITLE
Set up configuration for mailgun

### PR DESCRIPTION
**Pull request by Hoverbear**
Based off https://github.com/sozialhelden/wheelmap/pull/437.
Contributes to https://github.com/sozialhelden/wheelmap/issues/349.

**This requires changes to `config/secrets.yml` before deployment.** The changes required can be found in the `infra-rebuild` branch of the chef repositories.

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -65,17 +65,16 @@
...
   config.action_mailer.smtp_settings = {
-    address: 'mail.wheelmap.org',
-    enable_starttls_auto: false,
-    port: 25,
-    domain: 'wheelmap.org',
-    authentication: :login,
-    user_name: Rails.application.secrets.mail_user,
-    password: Rails.application.secrets.mail_password
-
+    :authentication => :plain,
```
Comment (1000miles) [inline config/environments/production.rb]: So when we are using `plain` for authentication shouldn't we then set `enable_starttls_auto: true`?

Comment (Hoverbear) [inline config/environments/production.rb]: I based the configuration off https://devcenter.heroku.com/articles/mailgun#sending-emails-via-smtp ... You are probably right though!

Comment (Hoverbear) [inline config/environments/production.rb]: > `:enable_starttls_auto` - Detects if STARTTLS is enabled in your SMTP server and starts to use it. Defaults to `true`.

It defaults to `true`: http://api.rubyonrails.org/v5.1.0/classes/ActionMailer/Base.html

File: config/environments/production.rb [hunk 2]
```diff
@@ -65,17 +65,16 @@
...
+    :address        => 'smtp.mailgun.org',
+    :port           => 587,
+    :domain         => 'mg.wheelmap.org',
```
Comment (1000miles) [inline config/environments/production.rb]: Why aren't we use env variables for `user_name` and `password` here?

File: config/environments/staging.rb [hunk 3]
```diff
@@ -62,17 +62,16 @@
...
-    port: 25,
-    domain: 'wheelmap.org',
-    authentication: :login,
-    user_name: Rails.application.secrets.mail_user,
-    password: Rails.application.secrets.mail_password
-
+    :authentication => :plain,
+    :address        => 'smtp.mailgun.org',
+    :port           => 587,
+    :domain         => 'mg.wheelmap.org',
```
Comment (1000miles) [inline config/environments/staging.rb]: Same here. Why aren't we use env variables for `user_name` and `password` here?

File: config/environments/production.rb (hunk 1 above)
Comment (1000miles) [inline config/environments/production.rb]: True. :)

File: config/environments/production.rb (hunk 2 above)
Comment (Hoverbear) [inline config/environments/production.rb]: We don't currently do it now. That would be a separate PR, perhaps?

Comment (1000miles) [inline config/environments/production.rb]: Why not in this PR? :)

Comment (Hoverbear) [inline config/environments/production.rb]: Because `config/secrets.yml` seems like an appropriate place to do it.

Comment (1000miles) [inline config/environments/production.rb]: Right, but how do you want to fetch the credentials after you put them into the secrets.yml file?

From what I understand so far we  basically put in `config/secrets.yml`  the credentials:

```ruby

# Production Mailgun

production:
  mail_username: <%= ENV["mail_username"] %>
  mail_password: <%= ENV["mail_password"] %>
```
And call it e.g. from `config/environment/production.rb`:

```ruby
  config.action_mailer.smtp_settings = {
    :user => ENV["mail_username"],
    :password => ENV["mail_password"],
    :authentication => :plain,
    :address        => 'smtp.mailgun.org',
    :port           => 587,
    :domain         => 'mg.wheelmap.org',
}
```

See also: https://github.com/rails/rails/tree/4-1-stable/actionmailer

Comment (1000miles) [inline config/environments/production.rb]: oops, i accidently put ruby embedded code styling (ERB), should be this as an example in `secrets.yml`:

```ruby

# Mailgun

production:
  mail_username: "mail_username"
  mail_password: "mail_password"

```

Comment (Hoverbear) [inline config/environments/production.rb]: Ah, right. I missed these lines somehow: https://github.com/sozialhelden/wheelmap/pull/437/files#diff-1d98f0039bbc6df92adee882ed99d993L78

**Comment by Hoverbear** [comment]
@1000miles I fixed up that issue.

**Comment by 1000miles** [comment]
Tested with staging credentials. Works! Thanks a lot! :) 📬

**Comment by Hoverbear** [comment]
Let's wait on merge until we are ready to consciously ready to push this to production since it requires some manual intervention.
